File: vb/demo/Ice/async/Publisher.vb
' **********************************************************************
'
' Copyright (c) 2003-2006 ZeroC, Inc. All rights reserved.
'
' This copy of Ice is licensed to you under the terms described in the
' ICE_LICENSE file included in this distribution.
'
' **********************************************************************

Imports System
Imports Demo

Module AsyncC

    Class Publisher
        Inherits Ice.Application

        Private Sub menu()
            Console.WriteLine("Enter /quit to exit.")
        End Sub

        Public Overloads Overrides Function run(ByVal args() As String) As Integer
            Dim queue As QueuePrx = QueuePrxHelper.checkedCast(communicator().propertyToProxy("Queue.Proxy"))
            If queue Is Nothing Then
                Console.Error.WriteLine("invalid proxy")
                Return 1
            End If

	    Console.WriteLine("Type a message and hit return to queue a message.")
            menu()

            Dim line As String = Nothing
            Try
                While 1
                    Console.Out.Write("==> ")
                    Console.Out.Flush()
                    line = Console.In.ReadLine()
                    If line Is Nothing Then
                        Exit Try
                    End If
		    If not line.Length = 0 Then
                        If line.Chars(0) = "/" Then
			    If line.Equals("/quit") Then
			        Exit While
			    Else
			        menu()
			    End If
                        Else
                            queue.add(line)
                        End If
		    End If
            End While
            Catch ex As System.Exception
                Console.Error.WriteLine(ex)
            End Try

            Return 0
        End Function
    End Class

    Public Sub Main(ByVal args() As String)
        Dim app As Publisher = New Publisher
        Dim status As Integer = app.Main(args, "config.client")
        System.Environment.Exit(status)
    End Sub
End Module

File: cs/demo/Ice/async/Publisher.cs
// **********************************************************************
//
// Copyright (c) 2003-2006 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

using System;
using Demo;

public class Publisher : Ice.Application
{
    private static void menu()
    {
        Console.Out.WriteLine("Enter /quit to exit.");
    }

    public override int run(string[] args)
    {
        QueuePrx queue = QueuePrxHelper.checkedCast(communicator().propertyToProxy("Queue.Proxy"));
        if(queue == null)
        {
            Console.Error.WriteLine("invalid proxy");
            return 1;
        }

	Console.Out.WriteLine("Type a message and hit return to queue a message.");
        menu();

        string line = null;
	try
	{
            while(true)
            {
                Console.Out.Write("==> ");
		Console.Out.Flush();
                line = Console.In.ReadLine();
                if(line == null)
                {
                    break;
                }
                if(line.Length != 0)
                {
		    if(line[0] == '/')
		    {
		        if(line.Equals("/quit"))
			{
			    break;
			}
		        menu();
		    }
		    else
		    {
		        queue.add(line);
		    }
                }
            }
	}
        catch(System.Exception ex)
        {
	    Console.Error.WriteLine(ex);
        }

        return 0;
    }

    public static void Main(string[] args)
    {
        Publisher app = new Publisher();
        int status = app.main(args, "config.client");
	if(status != 0)
	{
	    System.Environment.Exit(status);
	}
    }
}
